### PR TITLE
Implement dynamic padding

### PR DESCRIPTION
**Pull request by JayOrten**
This PR reworks the tokenization process, so that, instead of pre padding/truncating all tokenized sequences, it dynamically pads throughout training. This introduces flexibility for sequence length.

- Removes padding/truncation parameters from HF tokenization function
- Updates dataset iterator to work based on sequence length
- Adds padding function for data collation, for dynamic padding of sequence to sequence length.

### CODE REVIEW

File: src/dataset.py [hunk 1]
```diff
@@ -113,7 +126,32 @@ def __iter__(self):
...
+
+    def pad_to_longest(self, batch):
+        """
+        Collator function for padding sequences to longest in batch, during training
+        """
+        x, y_true = zip(*batch)
+
+        x_lengths = [len(line) for line in x]
+        max_x_length = max(x_lengths)
+        padded_x = [line + [self.pad_token_id] * (max_x_length - len(line)) for line in x]
```
Comment (KimballNJardine) [inline src/dataset.py]: perhaps call padded_x "x_padded" to have the name help imply padding is on the end.

File: src/dataset.py [hunk 2]
```diff
@@ -113,7 +126,32 @@ def __iter__(self):
...
+
+        y_true_lengths = [len(s) for s in y_true]
+        max_y_true_lengths = max(y_true_lengths)
+        padded_y_true = [line + [self.pad_token_id] * (max_y_true_lengths - len(line)) for line in y_true]
```
Comment (KimballNJardine) [inline src/dataset.py]: perhaps call padded_y_true "y_padded_true" to have the name help imply padding is on the end.

File: src/dataset.py [hunk 3]
```diff
@@ -113,7 +126,32 @@ def __iter__(self):
...
+
+        padded_x = torch.tensor(padded_x)
+        padded_y_true = torch.tensor(padded_y_true)
```
Comment (KimballNJardine) [inline src/dataset.py]: See comments above.

File: src/dataset.py [hunk 4]
```diff
@@ -113,7 +126,32 @@ def __iter__(self):
...
+
+        x_lengths = [len(line) for line in x]
+        max_x_length = max(x_lengths)
+        padded_x = [line + [self.pad_token_id] * (max_x_length - len(line)) for line in x]
+
+        y_true_lengths = [len(s) for s in y_true]
+        max_y_true_lengths = max(y_true_lengths)
+        padded_y_true = [line + [self.pad_token_id] * (max_y_true_lengths - len(line)) for line in y_true]
+
+        padded_x = torch.tensor(padded_x)
```
Comment (KimballNJardine) [inline src/dataset.py]: See comments above.

**Review by KimballNJardine — CHANGES_REQUESTED** [review]
See minor comments on variable names. Otherwise fine.

File: src/dataset.py (hunk 1 above)
Comment (JayOrten) [reply src/dataset.py]: Done

**Review by KimballNJardine — APPROVED** [review]
Looks good.
Approved.
